packages/venia-ui/lib/components/App/contextProvider.js

```diff
@@ -17,10 +16,10 @@ const contextProviders = [
     ToastContextProvider
 ];
 
-const AppContextProvider = ({ children }) => {
+const ContextProvider = ({ children }) => {
```

Comment:
Does the `PeregrineContextProvider` include all of the Redux state slice contexts? I'm having trouble piecing together all the things that wrap `App`.

Reply:
Yea, we had this discussion back when we made the global reduced provider and it was confusing then too :P The `PeregrineContextProvider` is meant to export a single wrapper that contains all the other context providers that we felt *essential* to Peregrine.

I think now that we're shifting those lines of what is "owned" by a developer (the storefront is where customization is made) we should reconsider where we define this reducer. IMO we should move it to `venia-concept` just like we have `createStore` in the storefront.

@jimbo does what I'm saying make sense? If so we should open an issue for this.

Reply:
@supernova-at Yes. `PeregrineContextProvider` is simply all of Peregrine's Redux slice context providers, composed into a single component.

This is how a storefront app is intended to import and expose core app state, so we wanted it to involve a minimum of boilerplate.

Reply:
> IMO we should move it to venia-concept just like we have createStore in the storefront.

It's good to bring that up.

The way we're exporting reducers and letting `venia-concept` combine them is actually counter to my intent. I wanted to eliminate Redux so that Venia wouldn't have to create a store at all; since we couldn't eliminate Redux yet, though, I had to let Venia create the store because it would've been too hard for Venia to customize a store created by Peregrine.

With context providers, there's no need to make Venia compose them. Peregrine needs some certainty over the structure of app state, and Venia doesn't need more opportunities to be misconfigured. In the future, we may export individual providers _in addition to_ this pre-composed one, but I don't foresee it being necessary.